edm4hep.yaml

```
@@ -74,8 +74,9 @@ components:
         - float omega // is the signed curvature of the track in [1/mm].
         - float Z0 // longitudinal impact parameter
         - float tanLambda // lambda is the dip angle of the track in r-z
+	- float time // time of the track
         - edm4hep::Vector3f referencePoint // Reference point of the track parameters, e.g. the origin at the IP, or the position  of the first/last hits or the entry point into the calorimeter.
-        - std::array<float, 15> covMatrix // lower triangular covariance matrix of the track parameters.  the order of parameters is  d0, phi, omega, z0, tan(lambda). the array is a row-major flattening of the matrix.
+        - std::array<float, 21> covMatrix // lower triangular covariance matrix of the track parameters.  the order of parameters is  d0, phi, omega, z0, tan(lambda), time. the array is a row-major flattening of the matrix.
```

This will shuffle around the indices of covMatrix and break code that is currently using it ( not that much, but still):
https://github.com/cepc/CEPCSW/search?q=covMatrix&type=code
https://github.com/search?q=covMatrix+org%3Akey4hep&type=code

What about splitting this into two arrays? The full covariance matrix has to be constructed anyway for practical applications, so it makes no difference if that is done from covMat or covMat + covMatTimed

```suggestion
        - std::array<float, 15> covMatrix // lower triangular covariance matrix of the track parameters.  the order of parameters is  d0, phi, omega, z0, tan(lambda). the array is a row-major flattening of the matrix.
        - std::array<float, 7> covMatrixTimed // covariance matrix entries of the time-related track parameters.  the order of parameters is  sigma_t^2, d0, phi, omega, z0, tan(lambda)
```

---

Actually as it is the lower triangle, it just adds indices at the end of the matrix, so not needed to split this in two, right?

---

I agree that we do not need this complication of splitting the matrix as the existing indices will be unchanged.

---

That's true of course, I overlooked the **lower** triangle part!